### PR TITLE
Gomatic stage patterns to message pull requests

### PR DESCRIPTION
**Pull request by macdiesel**
@edx/pipeline-team PTAL.

I will integrate these in to the edxapp pipelines once @cpennington 's atomic PR lands. #118

### CODE REVIEW

File: edxpipelines/patterns/stages.py [hunk 1]
```diff
@@ -1152,3 +1152,108 @@ def generate_create_branch(pipeline,
...
+
+
+def generate_message_prs_on_stage(pipeline,
+                                  org,
+                                  repo,
+                                  token,
+                                  base_sha,
+                                  head_sha,
+                                  stage_name=constants.MESSAGE_PR_STAGE_NAME,
+                                  manual_approval=False):
```
Comment (cpennington) [inline edxpipelines/patterns/stages.py]: Couldn't this just be one `generate_message_prs` that takes an argument telling it what message to send? Likewise for the task patterns.

Comment (cpennington) [inline edxpipelines/patterns/stages.py]: Or possibly that plus the name of the stage, I guess.

Comment (macdiesel) [inline edxpipelines/patterns/stages.py]: The tasks are consolidated in to a single function really with _generate_message_pull_requests_in_commit_range, the other methods are convenience for the caller. My thought process here was basically to have easy to call and assemble patterns for the end user assembling pipelines. 

Basically trying to let the person who will later assemble and use these not have to think too hard about what stage name and job name to pass in, what constants need to be used or what flags need to be set.

Comment (cpennington) [inline edxpipelines/patterns/stages.py]: Fair enough

**Comment by cpennington** [comment]
I'd prefer to not merge this until https://github.com/edx/tubular/pull/98 lands, so that we don't start posting different kinds of messages in the same release cycle.

File: edxpipelines/patterns/stages.py (hunk 1 above)
Comment (macdiesel) [inline edxpipelines/patterns/stages.py]: FWIW when integrating it became clear a single method would work better and I pulled your suggestion into the code.

**Comment by macdiesel** [comment]
@cpennington @doctoryes this is read for re-review.

**Pull request merged**